src/Mod/Fem/feminout/importCcxFrdResults.py

```diff
@@ -176,30 +215,35 @@ def make_result_mesh(result_name):
                 # fill Stats
                 res_obj = resulttools.fill_femresult_stats(res_obj)
 
-                # create a results pipeline if not already existing
-                pipeline_name = "Pipeline_" + results_name
-                pipeline_obj = doc.getObject(pipeline_name)
-                if pipeline_obj is None:
-                    pipeline_obj = ObjectsFem.makePostVtkResult(doc, res_obj, results_name)
-                    pipeline_visibility = True
-                    if analysis:
-                        analysis.addObject(pipeline_obj)
+                # if we have multiple results we delay the pipeline creation
+                if number_of_increments == 1:
+                    setupPipeline(doc, analysis, results_name, [res_obj])
                 else:
-                    if FreeCAD.GuiUp:
-                        # store pipeline visibility because pipeline_obj.load makes the
-                        # pipeline always visible
-                        pipeline_visibility = pipeline_obj.ViewObject.Visibility
-                    pipeline_obj.load(res_obj)
-                # update the pipeline
-                pipeline_obj.recomputeChildren()
-                pipeline_obj.recompute()
-                if FreeCAD.GuiUp:
-                    pipeline_obj.ViewObject.updateColorBars()
-                    # make results mesh invisible, will be made visible
-                    # later in task_solver_ccxtools.py
-                    res_obj.Mesh.ViewObject.Visibility = False
-                    # restore pipeline visibility
-                    pipeline_obj.ViewObject.Visibility = pipeline_visibility
+                    multistep_value.append(step_time)
+                    multistep_result.append(res_obj)
+
+            # we have collected all result objects, lets create the multistep result pipeline
+            if number_of_increments > 1:
+                # figure out type and unit
+                unit = FreeCAD.Units.Unit("")
+                description = "Unknown"
+                if result_analysis_type == "frequency":
+                    unit = FreeCAD.Units.Frequency
+                    description = "Eigenmode"
+                elif result_analysis_type == "buckling":
+                    description = "Buckling factor"
+                elif result_analysis_type == "thermomech":
+                    unit = FreeCAD.Units.TimeSpan
+                    description = "Timesteps"
+                elif result_analysis_type == "static":
+                    description = "Load factor"
```

Comment:
In the following suggestion, I've:
* changed if chain to python 3.10's match-case to make it easier to read
* removed default unit and description and instead explicitly write them in each case
* use existing default constructor for unit instead of parsing an empty string
```suggestion
                match result_analysis_type:
                    case "frequency":
                        unit = FreeCAD.Units.Frequency
                        description = "Eigenmode"
                    case "buckling":
                        unit = FreeCAD.Units.Unit()
                        description = "Buckling factor"
                    case "thermomech":
                        unit = FreeCAD.Units.TimeSpan
                        description = "Timesteps"
                    case "static":
                        unit = FreeCAD.Units.Unit()
                        description = "Load factor"
                    case _:
                        unit = FreeCAD.Units.Unit()
                        description = "Unknown"
```